src/parser/base.lisp

```diff
@@ -0,0 +1,342 @@
+(defpackage #:coalton-impl/parser/base
+  (:use
+   #:cl)
+  (:shadow
+   #:parse-error)
+  (:local-nicknames
+   (#:util #:coalton-impl/util))
+  (:export
+   #:identifier                         ; TYPE
+   #:identifier-p                       ; FUNCTION
+   #:get-line-from-index                ; FUNCTION
+   #:get-source-line-info               ; FUNCTION
+   #:get-nth-line                       ; FUNCTION
+   #:make-coalton-error-note            ; FUNCTION
+   #:make-coalton-error-help            ; FUNCTION
+   #:make-coalton-error                 ; FUNCTION
+   #:coalton-error                      ; FUNCTION
+   #:display-coalton-error              ; FUNCTION
+   #:parse-error                        ; CONDITION
+   ))
+
+(in-package #:coalton-impl/parser/base)
+
+(deftype identifier ()
+  '(and symbol (not boolean) (not keyword)))
+
+(defun identifier-p (x)
+  (typep x 'identifier))
+
+(defun get-line-from-index (file index)
+  "Get the line number corresponding to the character offset INDEX.
+
+Returns (VALUES LINE-NUM LINE-START-INDEX)"
+  (declare (type sb-sys::fd-stream file)
```

Comment:
I can change this. Regular streams don't always support `file-position`, and the clhs doesn't define a type for "streams that do support file-position."

I think the best way to avoid SBCL specific functionality would be to run CI against another lisp implementation. I'm not sure what that lisp implementation would be though.

Reply:
Pretty much the only option IMO is CCL unless you can convince Franz to get your hands on an Allegro CI setup.